.github/workflows/boulder-ci.yml

```diff
@@ -105,6 +105,44 @@ jobs:
       - name: "Run Test: ${{ matrix.tests }}"
         run: ${{ matrix.tests }}
 
+  govulncheck:
+    runs-on: ubuntu-20.04
+    strategy:
+      # When set to true, GitHub cancels all in-progress jobs if any matrix job fails. Default: true
+      fail-fast: false
+      matrix:
+        # Add additional docker image tags here and all tests will be run with the additional image.
+        BOULDER_TOOLS_TAG:
+          - go1.20.5_2023-06-20
+          - go1.21rc2_2023-06-21
+
+    env:
+      # This sets the docker image tag for the boulder-tools repository to
+      # use in tests. It will be set appropriately for each tag in the list
+      # defined in the matrix.
+      BOULDER_TOOLS_TAG: ${{ matrix.BOULDER_TOOLS_TAG }}
+
+    steps:
+      # Enable https://github.com/golang/go/wiki/LoopvarExperiment if we're on
+      # go1.21rc2 or higher. This experiment value is unknown in lower versions.
+      - if: startsWith(matrix.GO_VERSION, '1.21')
+        run: echo "GOEXPERIMENT=loopvar" >> "$GITHUB_ENV"
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Print the env variable being used to pull the docker image. For
+      # informational use.
+      - name: Print BOULDER_TOOLS_TAG
+        run: echo "Using BOULDER_TOOLS_TAG ${BOULDER_TOOLS_TAG}"
+
+      # Pre-pull the docker containers before running the tests.
+      - name: docker compose pull boulder
+        run: docker compose pull boulder
+
+      - name: Install and run govulncheck
+        run: docker compose run --use-aliases netaccess bash -c 'GOFLAGS="" GOBIN=/usr/local/bin go install golang.org/x/vuln/cmd/govulncheck@latest && govulncheck ./...'
```

Comment:
Can this be split into two steps?

Reply:
It can be, but then we have to worry about doing container cleanups and shutting the container down in a timely manner. Something like this which I don't recommend because `if: always()` [is _very weird_](https://github.com/orgs/community/discussions/26303). 
```
      - name: Run netaccess container
        run: docker compose run --name gvccontainer --use-aliases --detach netaccess sh -c "while true; do sleep 1; done"
        timeout-minutes: 2

      - name: Install govulncheck
        run: docker exec gvccontainer bash -c 'GOFLAGS="" GOBIN=/usr/local/bin go install golang.org/x/vuln/cmd/govulncheck@latest'

      - name: Run govulncheck
        run: docker exec gvccontainer govulncheck ./...

      - if: always()
        run: docker stop gvccontainer
```